### PR TITLE
Build kdl from source, fixing all kdl issues

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Test in ipython:
```
import PyKDL as kdl
A=kdl.Frame(kdl.Rotation.Quaternion(0,0,0,1), kdl.Vector(1,2,3))
B=kdl.Frame(kdl.Rotation.Quaternion(0,0,0,1), kdl.Vector(0,0,0))
C=(B*A)
print C.p
print (B*A).p
```
The last line will fail in the old situation

### CODE REVIEW

**Comment by Timple** [comment]
I did never install kdl from source to my knowledge. But your example does not crash at my pc...

File: installer/targets/ros-orocos_kdl/install.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 - type: ros
   source:
-    type: system
-    name: orocos-kdl
+      type: git
+      url: https://github.com/smart-robotics/orocos_kinematics_dynamics.git
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply installer/targets/ros-orocos_kdl/install.yaml]: Sorry, copy-paste....

**Comment by Timple** [comment]
Still cannot reproduce, not even in a docker image.

Can you try the same as I did below?

```
tim@Precision $ docker run -it tuerobotics/tue-env:master
amigo@a8ff1eb319e2:~$ sudo apt update -qq
88 packages can be upgraded. Run 'apt list --upgradable' to see them.
amigo@a8ff1eb319e2:~$ tue-get install ros-python_orocos_kdl 
Going to run the following command:

sudo apt-get install --assume-yes  ros-kinetic-python-orocos-kdl

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libeigen3-dev python-sip python-sip-dev ros-kinetic-orocos-kdl sip-dev
Suggested packages:
  libeigen3-doc libmrpt-dev python-sip-doc
The following NEW packages will be installed:
  libeigen3-dev python-sip python-sip-dev ros-kinetic-orocos-kdl ros-kinetic-python-orocos-kdl sip-dev
0 upgraded, 6 newly installed, 0 to remove and 88 not upgraded.
Need to get 1222 kB of archives.
After this operation, 7525 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu xenial/universe amd64 python-sip amd64 4.17+dfsg-1build1 [71.1 kB]
Get:2 http://archive.ubuntu.com/ubuntu xenial/universe amd64 sip-dev amd64 4.17+dfsg-1build1 [138 kB]
Get:3 http://packages.ros.org/ros/ubuntu xenial/main amd64 ros-kinetic-orocos-kdl amd64 1.3.1-0xenial-20180222-191455-0800 [238 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial/universe amd64 python-sip-dev amd64 4.17+dfsg-1build1 [19.9 kB]
Get:5 http://archive.ubuntu.com/ubuntu xenial/universe amd64 libeigen3-dev all 3.3~beta1-2 [663 kB]
Get:6 http://packages.ros.org/ros/ubuntu xenial/main amd64 ros-kinetic-python-orocos-kdl amd64 1.3.1-0xenial-20180222-192040-0800 [91.4 kB]
Fetched 1222 kB in 3s (383 kB/s)                       
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 6.)
debconf: falling back to frontend: Readline
Selecting previously unselected package python-sip.
(Reading database ... 67097 files and directories currently installed.)
Preparing to unpack .../python-sip_4.17+dfsg-1build1_amd64.deb ...
Unpacking python-sip (4.17+dfsg-1build1) ...
Selecting previously unselected package sip-dev.
Preparing to unpack .../sip-dev_4.17+dfsg-1build1_amd64.deb ...
Unpacking sip-dev (4.17+dfsg-1build1) ...
Selecting previously unselected package python-sip-dev.
Preparing to unpack .../python-sip-dev_4.17+dfsg-1build1_amd64.deb ...
Unpacking python-sip-dev (4.17+dfsg-1build1) ...
Selecting previously unselected package libeigen3-dev.
Preparing to unpack .../libeigen3-dev_3.3~beta1-2_all.deb ...
Unpacking libeigen3-dev (3.3~beta1-2) ...
Selecting previously unselected package ros-kinetic-orocos-kdl.
Preparing to unpack .../ros-kinetic-orocos-kdl_1.3.1-0xenial-20180222-191455-0800_amd64.deb ...
Unpacking ros-kinetic-orocos-kdl (1.3.1-0xenial-20180222-191455-0800) ...
Selecting previously unselected package ros-kinetic-python-orocos-kdl.
Preparing to unpack .../ros-kinetic-python-orocos-kdl_1.3.1-0xenial-20180222-192040-0800_amd64.deb ...
Unpacking ros-kinetic-python-orocos-kdl (1.3.1-0xenial-20180222-192040-0800) ...
Processing triggers for man-db (2.7.5-1) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Setting up python-sip (4.17+dfsg-1build1) ...
Setting up sip-dev (4.17+dfsg-1build1) ...
Setting up python-sip-dev (4.17+dfsg-1build1) ...
Setting up libeigen3-dev (3.3~beta1-2) ...
Setting up ros-kinetic-orocos-kdl (1.3.1-0xenial-20180222-191455-0800) ...
Setting up ros-kinetic-python-orocos-kdl (1.3.1-0xenial-20180222-192040-0800) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
amigo@a8ff1eb319e2:~$ import PyKDL as kdl
import: unable to open X server `' @ error/import.c/ImportImageCommand/364.
amigo@a8ff1eb319e2:~$ ipython
bash: ipython: command not found
amigo@a8ff1eb319e2:~$ python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import PyKDL as kdl
>>> A=kdl.Frame(kdl.Rotation.Quaternion(0,0,0,1), kdl.Vector(1,2,3))
>>> B=kdl.Frame(kdl.Rotation.Quaternion(0,0,0,1), kdl.Vector(0,0,0))
>>> C=(B*A)
>>> print C.p
[           1,           2,           3]
>>> print (B*A).p
[           0,           2,           3]
>>> 
```

**Comment by MatthijsBurgh** [comment]
@Timple you are pulling from master... Therefore it is correct, it fails.

File: installer/targets/ros-python-sip/install.bash [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+local SIP_version="4.15.5"
+
+if [ $(lsb_release -sc) == "trusty" ]
+then
+    tue-install-system python-sip-dev
+else
+
+    if command -v sip >/dev/null 2>&1
```
Comment (Rayman) [inline installer/targets/ros-python-sip/install.bash]: Why are you installing python-sip from source?

**Comment by Timple** [comment]
Oww, dude! The command doesn't fail. It just gave a wrong output.
I thought everything was fine because nothing crashed :)

File: installer/targets/ros-python-sip/install.bash (hunk 2 above)
Comment (MatthijsBurgh) [reply installer/targets/ros-python-sip/install.bash]: SIP is the 'source' of the problem. Or the combination of KDL with different SIP versions. Therefore we need a specific SIP version.

File: installer/targets/ros-python-tensorflow-pip/install.yaml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 - type: pip
-  name: tensorflow>=1.4.1
+  name: tensorflow-gpu>=1.4.1
```
Comment (Rayman) [inline installer/targets/ros-python-tensorflow-pip/install.yaml]: This belongs in a different PR

File: installer/targets/ros-python-sip/install.bash [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+local SIP_version="4.15.5"
```
Comment (Rayman) [inline installer/targets/ros-python-sip/install.bash]: You can also use 4.16.7, but this version is also fine

File: installer/targets/ros-python-sip/install.bash [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    fi
+    
+    tue-install-info "Installing SIP version: $SIP_version"
+    cp $SIP_file /tmp/sip.tar.gz
+
+    tar xvzf /tmp/sip.tar.gz -C /tmp
+    cd /tmp/sip-$SIP_version
+    python ./configure.py
+    make
+    sudo make install
```
Comment (Rayman) [inline installer/targets/ros-python-sip/install.bash]: Please don't use `sudo make install` because it makes it impossible to remove the package again. Use the following:
```
sudo checkinstall -y --nodoc make install
```

File: installer/targets/ros-python-tensorflow-pip/install.yaml (hunk 3 above)
Comment (MatthijsBurgh) [reply installer/targets/ros-python-tensorflow-pip/install.yaml]: Oops

File: installer/targets/ros-python-sip/install.bash (hunk 5 above)
Comment (MatthijsBurgh) [reply installer/targets/ros-python-sip/install.bash]: When using ```sudo checkinstall -y --nodoc make install``` the following log shows:
```
(Reading database ... 469674 files and directories currently installed.)
Preparing to unpack .../sip_4.15.5-1_amd64.deb ...
Unpacking sip (4.15.5-1) ...
dpkg: error processing archive /tmp/sip-4.15.5/sip_4.15.5-1_amd64.deb (--install):
 trying to overwrite '/usr/bin/sip', which is also in package sip-dev 4.17+dfsg-1build1
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Errors were encountered while processing:
 /tmp/sip-4.15.5/sip_4.15.5-1_amd64.deb
```
The log is shown in a editor, so you have to close that one to continue the installer.

But afterwards ```sip -V``` does provide the correct version.

Comment (MatthijsBurgh) [reply installer/targets/ros-python-sip/install.bash]: @Rayman I don't see a correct use of `checkinstall`. Do know any solution?

Comment (Rayman) [reply installer/targets/ros-python-sip/install.bash]:
```
 trying to overwrite '/usr/bin/sip', which is also in package sip-dev 4.17+dfsg-1build1
```
This is good, dpkg keeps track of files so when you uninstall packages, all the files are removed. You are not able to uninstall sip when you did `sudo make install`

Comment (MatthijsBurgh) [reply installer/targets/ros-python-sip/install.bash]: But it is already overwritten by `checkinstall`:
```
make[1]: Entering directory '/tmp/sip-4.15.5/sipgen'
cp -f sip /usr/bin/sip
```
Therefore the use of `checkinstall` is not really usefull

File: installer/targets/ros-python-tensorflow-pip/install.yaml (hunk 3 above)
Comment (Rayman) [reply installer/targets/ros-python-tensorflow-pip/install.yaml]: @MatthijsBurgh, why didn't you fix my comment? This would have prevented the fixup from yesterday.

Comment (MatthijsBurgh) [reply installer/targets/ros-python-tensorflow-pip/install.yaml]: I probably forgot, It wasn't my intention.
